### PR TITLE
allow sha-1 password during convert to shib #3287

### PR DESCRIPTION
**Pull request by pdurbin**
# RFI Checklist

### 1. Related Issues

- #3287 Upgraded (Bcrypt) password required for conversion from local to Shibboleth account 

---
### 2. Pull Request Checklist

- [x]  Functionality completed as described in FRD (no FRD)
- [x]  Dependencies, risks, assumptions in FRD addressed (no FRD)
- [x]  Unit tests completed (REST Assured tests)
- [x]  Deployment requirements identified (e.g., SQL scripts, indexing) (none)
- [x]  Documentation completed (none)
- [x]  All code checkins completed (none)

---
### 3. Review Checklist

_**After** the pull request has been submitted, fill out this section._

- [x]  Code review completed or waived
- [x]  Testing requirements completed
- [ ]  Usability testing completed or waived
- [ ]  Support testing completed or waived
- [ ]  Merged with develop branch and resolved conflicts

### CODE REVIEW

**Comment by pdurbin** [comment]
@kcondon @akio-sone @donsizemore I thought it might be helpful for you to see how I tested this locally.

Step 1 is to create a local account with an email address that you know will match the email address that the Shib IdP will assert (this is how the conversion process is kicked off). I used "ghopper@mailinator.com".

Step 2 is to set `passwordencryptionversion` to "0" and `encryptedpassword` to a known password. `4G7xxL9z11/JKN4jHPn4g9iIQck=` is "sha-1Pass". (Hat tip to @michbarsinai for the info at https://github.com/IQSS/dataverse/pull/2922#issuecomment-190818688 .) Here's how it looks:

```
(dataverse) murphy:dataverse pdurbin$ psql dataverse_db -c 'select email,encryptedpassword,passwordencryptionversion from builtinuser where id = 213;'
         email          |      encryptedpassword       | passwordencryptionversion 
------------------------+------------------------------+---------------------------
 ghopper@mailinator.com | 4G7xxL9z11/JKN4jHPn4g9iIQck= |                         0
(1 row)
```

Then, you log in with Shib. (In dev, I used the curl command below to get "ghopper@mailinator.com" from the Shib IdP but this wont work on a real server: `curl http://localhost:8080/api/admin/settings/:DebugShibAccountType -X PUT -d HARVARD2`)

You should be prompted for to convert. Here are the screenshots I saw:

screen shot 2016-08-17 at 12 09 17 pm

screen shot 2016-08-17 at 12 09 38 pm

This is the expected output in the logs:

```
[2016-08-17T12:09:26.596-0400] [glassfish 4.1] [INFO] [] [edu.harvard.iq.dataverse.authorization.providers.shib.ShibServiceBean] [tid: _ThreadID=93 _ThreadName=http-listener-1(3)] [timeMillis: 1471450166596] [levelValue: 800] [[
  The username and/or password entered is invalid: passwordreset.xhtml?token=093da2c2-1e39-40b4-bcc2-b95eee69863c&faces-redirect=true]]

[2016-08-17T12:09:26.597-0400] [glassfish 4.1] [INFO] [] [edu.harvard.iq.dataverse.authorization.providers.shib.ShibServiceBean] [tid: _ThreadID=93 _ThreadName=http-listener-1(3)] [timeMillis: 1471450166597] [levelValue: 800] [[
  AuthenticationFailedException caught in canLogInAsBuiltinUser: The username and/or password entered is invalid: passwordreset.xhtml?token=093da2c2-1e39-40b4-bcc2-b95eee69863c&faces-redirect=true - Maybe the user (ghopper) hasn't upgraded their password? Checking the old password...]]

[2016-08-17T12:09:26.601-0400] [glassfish 4.1] [INFO] [] [edu.harvard.iq.dataverse.authorization.AuthenticationServiceBean] [tid: _ThreadID=93 _ThreadName=http-listener-1(3)] [timeMillis: 1471450166601] [levelValue: 800] [[
  converting user 221 from builtin to shib]]
```

**Comment by pdurbin** [comment]
Oh, this issue and fix is highly related to #1034 about bcrypt which @michbarsinai worked on so if he'd like to do some code review, that would be awesome.

**Comment by pdurbin** [comment]
@kcondon please see https://help.hmdc.harvard.edu/Ticket/Display.html?id=239603 for more information on testing by @jonc1438 @akio-sone and @donsizemore .

File: src/main/java/edu/harvard/iq/dataverse/api/Admin.java [hunk 1]
```diff
@@ -416,7 +419,11 @@ public Response builtin2shib(String content) {
             if (!knowsExistingPassword) {
                 String message = "User doesn't know password.";
                 problems.add(message);
-                return errorResponse(Status.BAD_REQUEST, message);
+                /**
+                 * @todo Someday we should make a errorResponse method that
+                 * takes JSON arrays and objects.
+                 */
+                return errorResponse(Status.BAD_REQUEST, problems.build().toString());
```
Comment (michbarsinai) [inline src/main/java/edu/harvard/iq/dataverse/api/Admin.java]: I agree - you can add one `AbstractApiBean#okResponse( JsonObjectBuilder bad )`

**Comment by pdurbin** [comment]
@michbarsinai indicated to me that he check the box that he's done with code review (looks fine).

**Comment by kcondon** [comment]
OK, passed testing and code review. Merging.

**Pull request merged**